zxlive/commands.py

```diff
@@ -120,6 +145,8 @@ def undo(self) -> None:
     def redo(self) -> None:
         self._old_etys = [self.g.edge_type(e) for e in self.es]
         for e in self.es:
+            if self.g.edge_type(e) == EdgeType.W_IO:
+                continue
```

Comment:
Note sure if this check belongs here. I would expect a command that changes edge colors to actually affect all specified edges. Maybe we should ignore IO edges before invoking the command?

Reply:
The W_IO edge is not a real edge, as in it does not have any semantics so it never makes sense to change a W_IO edge to any other type. But I am happy to move this check to _ety_clicked method instead.

Reply:
Moving it to _ety_clicked is actually better because it skips adding empty command to the undo stack. Now the ChangeEdgeType command does not have this check: 843bdb667890bc59c1470b9ded6c12e538176133